mountpoint-s3-fs/src/config.rs

```diff
@@ -30,13 +30,13 @@ impl MountpointConfig {
             fuse_session_config,
             data_cache_config,
             filesystem_config,
-            error_callback: None,
+            error_logger: None,
         }
     }
 
-    /// Set the [Self::error_callback] field
-    pub fn error_callback(mut self, error_callback: Option<Arc<dyn ErrorCallback + Send + Sync>>) -> Self {
-        self.error_callback = error_callback;
+    /// Set the [Self::error_logger] field
+    pub fn error_logger(mut self, error_logger: Option<Box<dyn ErrorLogger + Send + Sync>>) -> Self {
```

Comment:
Do we need the `Option` here?

Would something like this work?
```suggestion
    pub fn error_logger(mut self, error_logger: impl ErrorLogger + Send + Sync) -> Self {
```